snap/snapcraft.yaml

```diff
@@ -64,17 +64,222 @@ parts:
     prime:
       - -*
 
+  buildenv:
+    plugin: nil
+    build-environment: &buildenv
+      - ACLOCAL_PATH: $CRAFT_STAGE/usr/share/aclocal
+      - XDG_DATA_DIRS: $CRAFT_STAGE/usr/share:/usr/share
+      - LD_LIBRARY_PATH: "$CRAFT_STAGE/usr/lib:\
+                          $CRAFT_STAGE/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR\
+                          ${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+      - PKG_CONFIG_PATH: "$CRAFT_STAGE/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/pkgconfig:\
+                          $CRAFT_STAGE/usr/lib/pkgconfig:\
+                          $CRAFT_STAGE/usr/share/pkgconfig\
+                          ${PKG_CONFIG_PATH:+:$PKG_CONFIG_PATH}"
+      - GI_TYPELIB_PATH: "$CRAFT_STAGE/usr/lib/girepository-1.0:\
+                          $CRAFT_STAGE/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/girepository-1.0\
+                          ${GI_TYPELIB_PATH:+:$GI_TYPELIB_PATH}"
+
+  ninja:
+    plugin: nil
+    source: https://github.com/ninja-build/ninja.git
+    source-tag: 'v1.12.1'
+    source-depth: 1
+    override-build: |
+      rm -rf build
+      rm -f ninja
+      rm -f ninja_bootstrap
+      sed -i 's_^#!/usr/bin/env python$_#!/usr/bin/env python3_g' configure.py
+      ./configure.py --bootstrap
+      mv ninja ninja_bootstrap
+      rm -rf build
+      ./ninja_bootstrap
+      rm -f ninja_bootstrap
+      mkdir -p $CRAFT_PART_INSTALL/usr/bin
+      mv ninja $CRAFT_PART_INSTALL/usr/bin/
+    build-environment: *buildenv
+    build-packages:
+      - python3
+    prime:
+      - -*
+
+  meson-deps:
+    after: [ ninja ]
+    plugin: nil
+    source: https://github.com/mesonbuild/meson.git
+    source-tag: '1.6.0'
```

Comment:
Same question, noting that `meson` 1.5 is in `noble-updates`?

Reply:
it's possible that meson in noble-updates might work today, but it could stop working pretty quickly as gtk updates, so this makes it trivial to build app the components that gtk4 needs just by bumping tags as needed.